apis/v1alpha2/shared_types.go

```diff
@@ -52,6 +52,11 @@ type ParentRef struct {
 	// Namespace is the namespace of the referent. When unspecified (or empty
 	// string), this refers to the local namespace of the Route.
 	//
+	// Note that when a namespace is specified, a ReferencePolicy object
+	// is required in the referent namespace to allow that namespace's
+	// owner to accept the reference. See the ReferencePolicy documentation
+	// for details.
```

Comment:
I think this is a bit more nuanced, maybe something like this would work?
```suggestion
	// Note that when a different namespace is specified, something in the
	// target namespace needs to allow that reference. When referring to a
	// Gateway in a different namespace, that Gateway needs to explicitly
	// allow for Routes to be attached from this namespace. For other kinds
	// of parent resources that don't include this configuration, a
	// ReferencePolicy must be present to accept the reference. See the 
	// ReferencePolicy documentation for details.
```